nmt.go

```diff
@@ -644,11 +644,37 @@ func (n *NamespacedMerkleTree) updateMinMaxID(id namespace.ID) {
 	}
 }
 
-type leafRange struct {
-	// start and end denote the indices of a leaf in the tree. start ranges from
-	// 0 up to the total number of leaves minus 1 end ranges from 1 up to the
-	// total number of leaves end is non-inclusive
-	start, end int
+// ComputeSubtreeRoot takes a leaf range and returns the corresponding subtree root.
+// Also, it requires the start and end range to correctly reference an inner node.
+func (n *NamespacedMerkleTree) ComputeSubtreeRoot(start, end int) ([]byte, error) {
+	if start < 0 {
+		return nil, fmt.Errorf("start %d shouldn't be strictly negative", start)
+	}
+	if end <= start {
+		return nil, fmt.Errorf("end %d should be stricly bigger than start %d", end, start)
+	}
+	uStart, err := safeIntToUint(start)
+	if err != nil {
```

Comment:
what do you mean its limits? currently, all possible errors are covered by the tests, if I missed any, please let me know.

Additionally, we have this test: https://github.com/celestiaorg/nmt/pull/260#discussion_r1655417247 that goes over all combinations, but it takes 4 min to execute. That's why I didn't add it.

Reply:
The codevoc complains about some of the lines in the code as not being covered by the tests, that's why I suggested expanding the tests to cover all execution paths. 
![Screenshot 2024-07-05 at 10 34 44 AM](https://github.com/celestiaorg/nmt/assets/35961250/636ab76c-1957-4b5f-a1c1-75ef00075128)

Reply:
this one is fine since the error is already tested for in the `safeIntToUint` tests. At this level, it should never occur. Same for the other lines reported by codecov